src/main/java/com/spotinst/sdkjava/model/SpotinstElastigroupService.java

```diff
@@ -1580,4 +1580,63 @@ public static Boolean deleteVolumeInStatefulInstance(String groupId, String stat
 
     }
 
+    public static ApiElastigroup importASG(ElastigroupImportASG request, String asgName, String dryRun, String region,
+                                                   String authToken, String account) {
+
+        ApiElastigroup retVal = null;
+
+        // Get endpoint
+        SpotinstHttpConfig config      = SpotinstHttpContext.getInstance().getConfiguration();
+        String             apiEndpoint = config.getEndpoint();
+
+        // Build query params
+        Map<String, String> queryParams = new HashMap<>();
+
+        // Add account Id Query param
+        if (account != null) {
+            queryParams.put("accountId", account);
+        }
+
+        // Add auto scaling group name Query param
+        if (asgName != null) {
+            queryParams.put("autoScalingGroupName", asgName);
+        }
+
+        // Add dry run Query param
+        if (dryRun != null) {
+            queryParams.put("dryRun", dryRun);
+        }
+
+        // Add region Query param
+        if (region != null) {
+            queryParams.put("region", region);
+        }
+
+        // Get the headers
+        Map<String, String> headers = buildHeaders(authToken);
+
+        // Build URI
+        String uri = String.format("%s/aws/ec2/group/autoScalingGroup/import", apiEndpoint);
+
+        // Write to json
+        Map<String, ElastigroupImportASG> groupRequest = new HashMap<>();
```

Comment:
This area seems problematic. Please recheck. I didn't get how parameters passed to this function are getting converted to group { ... } JSON.